trio/_dtls.py

```diff
@@ -380,13 +407,17 @@ def decode_volley_trusted(volley):
 
 
 class RecordEncoder:
-    def __init__(self):
+    def __init__(self) -> None:
```

Comment:
IIRC this isn't required? Is this cause mypy requires that some part of the signature be type-annotated in order to type-check the contents?

Reply:
Yeah, at least one part needs to be. Probably easiest to just always annotate.

Reply:
Unless we enable Mypy's check-untyped-defs option, which is usually more trouble than it's worth in this kind of migration.

Reply:
yeah `check_untyped_defs` on the whole codebase is a long way away, but forbidding untyped function signatures will prompt to add the `-> None` where necessary on `__init__`s (as will pyright `--verifytypes`). IDR if you can require always specify `-> None` but I don't think it's much of a problem with status quo

Reply:
Yeah I think this is fine, just was wondering cause this is different than what you've done before (... at least that I noticed)